### PR TITLE
Add file.tidied similar to Puppet's tidy resource.

### PR DESCRIPTION
**Pull request by dhs-rec**
### What does this PR do?
Adds file.tidied as inspired by Puppet
### What issues does this PR fix or reference?
Implements #46752
### Tests written?
No
### Commits signed with GPG?
No

### CODE REVIEW

**Comment by cachedout** [comment]
I left some comments in #46752 on this.

File: salt/states/file.py [hunk 1]
```diff
@@ -1532,6 +1533,128 @@ def absent(name,
     return ret
 
 
+# Helper to match a given name against one or more glob patterns
+def _matches(name, patterns=[]):
+    for pattern in patterns:
+        if fnmatch.fnmatch(name, pattern):
```
Comment (gtmanfred) [inline salt/states/file.py]: I would use regex instead of fnmatch

**Review by gtmanfred — CHANGES_REQUESTED** [review]

File: salt/states/file.py (hunk 1 above)
Comment (dhs-rec) [reply salt/states/file.py]: Done.

**Review by gtmanfred — APPROVED** [review]

**Comment by rallytime** [comment]
@dhs-rec Can you write some tests for this new function?

**Comment by dhs-rec** [comment]
Once I manage to wrap my mind around the test stuff...

**Comment by rallytime** [comment]
@dhs-rec Awesome! Thank you!

There's some docs on testing [here](https://docs.saltstack.com/en/latest/topics/development/tests/index.html), if it helps.

**Comment by rallytime** [comment]
Hi @dhs-rec - just a bump on this one. Were you able to write some tests for this?

**Comment by dhs-rec** [comment]
No, sorry. Tried last week but I somehow didn't get it (I'm not that experienced in Python hacking). So I would be glad if someone else could jump in...

**Comment by rallytime** [comment]
@gtmanfred Do you think you could help out with some tests on this?

**Comment by gtmanfred** [comment]
Yeah, make an issue and i will work on them :+1:

**Comment by gtmanfred** [comment]
@dhs-rec i have opened a PR on your branch, once you merge it we should be able to get this PR in.

https://github.com/dhs-rec/salt/pull/1

Thanks,
Daniel

**Comment by dhs-rec** [comment]
@gtmanfred: Great, thanks a lot!

**Pull request merged**